tests/ops/test_qubit_ops.py

```diff
@@ -1410,3 +1410,189 @@ def f2():
         state_2 = f2()
 
         assert np.allclose(state_1, state_2)
+
+    @pytest.mark.parametrize(
+        "control_wires,wires,control_values,expected_error_message",
+        [
+            ([0, 1], 2, 4, "Not enough control wires."),
+            ([0, 1], 2, "ab", "String of control values can contain only '0' or '1'."),
+            ([0, 1], 2, "011", "Length of control bit string must equal number of control wires."),
+            ([0, 1], 2, [0, 1], "Alternative control values must be passed as integer"),
+        ],
+    )
+    def test_invalid_mixed_polarity_controls(
+        self, control_wires, wires, control_values, expected_error_message
+    ):
+        """Test if ControlledQubitUnitary properly handles invalid mixed-polarity
+        control values."""
+        target_wires = Wires(wires)
+
+        dev = qml.device("default.qubit", wires=len(control_wires + target_wires))
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.ControlledQubitUnitary(
+                X, control_wires=control_wires, wires=target_wires, control_values=control_values
+            )
+            return qml.state()
+
+        with pytest.raises(ValueError, match=expected_error_message):
+            circuit()
+
+    @pytest.mark.parametrize(
+        "control_wires,wires,control_values",
+        [
+            ([0], 1, 0),
+            ([0, 1], 2, 0),
+            ([0, 1], 2, "01"),
+            ([0, 1], 2, 2),
+            ([0, 1], 2, "11"),
+            ([1, 0], 2, "10"),
+            ([0, 1], [2, 3], 3),
+            ([0, 2], [3, 1], 2),
+            ([1, 2, 0], [3, 4], "100"),
+            ([1, 0, 2], [4, 3], 6),
+        ],
+    )
+    def test_mixed_polarity_controls(self, control_wires, wires, control_values):
+        """Test if ControlledQubitUnitary properly applies mixed-polarity
+        control values."""
+        target_wires = Wires(wires)
+
+        dev = qml.device("default.qubit", wires=len(control_wires + target_wires))
+
+        # Pick a random unitary
+        U = unitary_group.rvs(2 ** len(target_wires), random_state=3)
```

Comment:
```suggestion
        U = unitary_group.rvs(2 ** len(target_wires), random_state=1967)
```
I like this seed because it's when PL was released! But either way is fine :laughing: